### PR TITLE
ROC-3567 Refactor 'mobile' out of the code-base

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3567

### Change description
Refactor out mentions of "mobile" phones, replacing them with something else appropriate.

Backend PR: https://github.com/hmcts/cmc-claim-store/pull/1049

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
I will need to add tests that should have been there anyway so that this PR has over 80% coverage.

File: src/main/common/utils/freeMediationUtil.ts [hunk 1]
```diff
@@ -35,29 +34,14 @@ export class FreeMediationUtil {
...
+          return claim.claimData.claimant.phoneNumber || mediationDraft.canWeUse.mediationPhoneNumber
         }
       } else {
         return mediationDraft.canWeUse.mediationPhoneNumber
       }
     }
     return undefined
   }
-
-  static getMediationContactPerson (claim: Claim, mediationDraft: MediationDraft, draft?: ResponseDraft): string {
```
Comment (robertparkinson) [inline src/main/common/utils/freeMediationUtil.ts]: why is this being removed?

File: src/main/app/claims/claimModelConverter.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ export class ClaimModelConverter {
           individualDetails.name,
           this.convertAddress(individualDetails.address),
           individualDetails.hasCorrespondenceAddress ? this.convertAddress(individualDetails.correspondenceAddress) : undefined,
-          draftClaim.claimant.mobilePhone.number,
+          draftClaim.claimant.phone.number,
```
Comment (robertparkinson) [inline src/main/app/claims/claimModelConverter.ts]: can this be done in two stages please? have the converter check for the old version and use it is it's there and if not then use the new one. As the phone number is optional if we do this all the phone numbers in the drafts will be lost and the users won't notice.

File: src/main/common/utils/freeMediationUtil.ts (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/common/utils/freeMediationUtil.ts]: Unused

File: src/main/app/claims/claimModelConverter.ts (hunk 2 above)
Comment (andrewwa-kainos) [reply src/main/app/claims/claimModelConverter.ts]: Agreed, and noting that I had asked you about this previously and had confirmation to go ahead regardless.

Comment (andrewwa-kainos) [reply src/main/app/claims/claimModelConverter.ts]: 6d6454ad5a1afe432338388f8349b27fe46d88ec

File: src/main/app/claims/models/details/yours/party.ts [hunk 3]
```diff
@@ -32,7 +34,10 @@ export class Party {
       this.name = input.name
       this.type = input.type
       this.email = input.email
-      this.mobilePhone = input.mobilePhone
+      this.phoneNumber = input.phoneNumber
+      if (!this.phoneNumber && input.mobilePhone) {
+        this.phoneNumber = input.mobilePhone
+      }
```
Comment (andrewwa-kainos) [inline src/main/app/claims/models/details/yours/party.ts]: would `this.phoneNumber = input.phoneNumber || input.mobilePhone` do all this?

File: src/main/common/utils/freeMediationUtil.ts (hunk 1 above)
Comment (robertparkinson) [reply src/main/common/utils/freeMediationUtil.ts]: it is used on the mediation branch. I pulled it out into a PR to make that one more manageable

File: src/main/app/claims/claimModelConverter.ts (hunk 2 above)
Comment (robertparkinson) [reply src/main/app/claims/claimModelConverter.ts]: I must have been having a stressful day and hadn't considered the fact that it was optional. if it was a mandatory field then the users would be notified and be able to put it in after.

File: src/main/common/utils/freeMediationUtil.ts (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/common/utils/freeMediationUtil.ts]: Re-added

**Pull request merged**